hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -340,6 +333,12 @@ private[hudi] object HoodieSparkSqlWriter {
     if (operation != WriteOperationType.DELETE) {
       if (mode == SaveMode.ErrorIfExists && tableExists) {
         throw new HoodieException(s"hoodie table at $tablePath already exists.")
+      } else if (mode == SaveMode.Overwrite && tableExists && operation !=  WriteOperationType.INSERT_OVERWRITE_TABLE) {
```

Comment:
this is inline with https://github.com/apache/hudi/blob/b6d363c0d4a6906ef1c9ea54c0263c6b5915e0b6/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala#L301